src/software/jetson_nano/services/motor.h

```diff
@@ -45,15 +47,58 @@ class MotorService : public Service
         const TbotsProto::DirectControlPrimitive_DirectVelocityControl& local_velocity,
         float dribbler_speed_rpm);
 
+    /**
+     * Trinamic API binding, sets spi_cs_driver_to_controller_demux appropriately
+     * and calls readWriteByte. See C++ implementation file for more info
+     *
+     * @param motor Which motor to talk to (in our case, the chip select)
+     * @param data The data to send
+     * @param last_transfer The last transfer of uint8_t data for this transaction.
+     */
+    uint8_t tmc4671ReadWriteByte(uint8_t motor, uint8_t data, uint8_t last_transfer);
+    uint8_t tmc6100ReadWriteByte(uint8_t motor, uint8_t data, uint8_t last_transfer);
```

Comment:
curious why the interface for the motor service is at the byte level?

Reply:
discussed offline, i was going to update this PR with an example of the API usage, but I will include that in a later PR

this is just the glue that binds the trinamic api to the linux jetson nano spi

see extlibs/trinamic/tmc/ic/tmc4671/... for more info